lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -1245,3 +1245,87 @@ bool ForwardingOperand::visitForwardedValues(
     return visitor(args[0]);
   });
 }
+
+void swift::findTransitiveReborrowBaseValuePairs(
+    BorrowingOperand initialScopedOperand, SILValue origBaseValue,
+    llvm::DenseMap<SILPhiArgument *, BaseValueSet> &reborrowToBaseValuesMap,
+    function_ref<void(SILPhiArgument *, SILValue)> visitReborrowBaseValuePair) {
+  // We need a SetVector to make sure we don't revisit the same reborrow operand
+  // again.
+  SmallSetVector<std::tuple<Operand *, SILValue>, 4> worklist;
+
+  // Populate the worklist with reborrow and the base value
+  initialScopedOperand.visitScopeEndingUses([&](Operand *op) {
+    if (op->getOperandOwnership() == OperandOwnership::Reborrow) {
+      worklist.insert({op, origBaseValue});
+    }
+    return true;
+  });
+
+  // Size of worklist changes in this loop
+ for (unsigned idx = 0; idx < worklist.size(); idx++) {
+    Operand *reborrowOp;
+    SILValue baseValue;
+    std::tie(reborrowOp, baseValue) = worklist[idx];
+
+    auto borrowingOperand = BorrowingOperand::get(reborrowOp);
```

Comment:
Can you get rid of the ::get here? There were a bunch of new APIs Andy put up. I would look at seeing if some of them can help you here.

Reply:
Got rid of ::get. I don't think the new api's help in following the phiArg/baseValue pair. But I can rewrite `visitTransitiveEndBorrows` using the new api's I think.